src/Executor/Setup.cs

```diff
@@ -0,0 +1,290 @@
+        {
+            if (Tools.ContainsKey(commandToExecute.ToolName))
+            {
+                if(commandToExecute.ToolName.Equals("msbuild"))
+                {
+                    return Path.GetFullPath(Path.Combine(configPath, os.Equals("windows") ? Tools[commandToExecute.ToolName].Run["windows"] : Tools[commandToExecute.ToolName].Run["unix"]));
+                }
+                else if (commandToExecute.ToolName.Equals("console"))
+                {
+                    string extension = os.Equals("windows") ? Tools[commandToExecute.ToolName].Run["windows"] : Tools[commandToExecute.ToolName].Run["unix"];
+                    return Path.GetFullPath(Path.Combine(configPath, string.Format("{0}.{1}", commandToExecute.LockedSettings["Project"],extension)));
+                }
+            }
+            return string.Empty;
+        }
+            
+        public string FormatSetting(string option, string value, string type, string toolName)
+        {
+            string commandOption = null;
+            if (type.Equals("passThrough"))
+            {
+                commandOption = string.Format(" {0}", toolName.Equals("console") ? "" : value);
+            }
+            else
+            {
+                if (Tools.ContainsKey(toolName) && !string.IsNullOrEmpty(type))
+                {
+                    commandOption = Tools[toolName].ValueTypes[type];
+                    commandOption = commandOption.Replace("{name}", option).Replace("{value}", value);
+                }
+            }
+            return commandOption;
+        }
+
+        public string GetHelpCommand(string commandName)
+        {
+            if(Commands.ContainsKey(commandName))
+            {
+                Command commandToPrint = Commands[commandName];
+                StringBuilder sb = new StringBuilder();
+                Dictionary<string, string> commandParametersToPrint = new Dictionary<string, string>();
+                string value;
+                sb.Append("  Locked Settings (values can't be overwritten): ").AppendLine();
+                foreach(KeyValuePair<string, string> lockedSetting in commandToPrint.LockedSettings)
+                {
+                    value = lockedSetting.Value.Equals("default") ? FindSettingValue(lockedSetting.Key) : lockedSetting.Value;
+                    sb.Append(string.Format("    {0} ({1})= {2}", lockedSetting.Key, FindSettingType(lockedSetting.Key), value)).AppendLine();
+                    commandParametersToPrint[lockedSetting.Key] = value;
+                }
+
+                sb.AppendLine().Append("  Other Settings (values can be overwritten): ").AppendLine();
+                foreach (KeyValuePair<string, string> optionalSettings in commandToPrint.Settings)
+                {
+                    value = optionalSettings.Value.Equals("default") ? FindSettingValue(optionalSettings.Key) : optionalSettings.Value;
```

Comment:
I had it as null before but after talking to Mark it was not clear that it was actually referring to the default value.
I could change it to `{default}`. That is unlikely to happen.